pandas-stubs/core/dtypes/missing.pyi

```diff
@@ -20,9 +23,9 @@ def isna(obj: DataFrame) -> DataFrame: ...
 @overload
 def isna(obj: Series) -> Series[bool]: ...
 @overload
-def isna(obj: Index | list | ArrayLike) -> np.ndarray: ...
+def isna(obj: Index | list | ArrayLike) -> npt.NDArray[np.bool_]: ...
 @overload
-def isna(obj: Scalar) -> bool: ...
+def isna(obj: Scalar | NaTType | NAType) -> bool: ...
```

Comment:
Probably not important: Could have an overload specific for `NaTType/NAType` as they should return `True`: `def isna(obj: NaTType | NAType) -> Literal[True]: ...`